src/dbnode/storage/types.go

```diff
@@ -278,6 +278,14 @@ type databaseNamespace interface {
 		opts index.QueryOptions,
 	) (index.QueryResults, error)
 
+	// AggregateQuery resolves the given query into unique tag/value pairs for
+	// that query.
+	AggregateQuery(
+		ctx context.Context,
+		query index.Query,
+		opts index.QueryOptions,
```

Comment:
probably need an `AggregateQueryOptions` struct

Reply:
Yeah, good point; was initially toying with using some of the existing functions and rolling specific `AggregateQueryOptions` and `AggregateResults` into `QueryOptions` and `Results` respectively, but less of a concern after refactoring